distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso

```diff
@@ -3,67 +3,62 @@ from Standard.Base import all
 ## UNSTABLE
    ADVANCED
 
-   A meta-representation of a runtime value.
+   An Atom meta-representation.
 
-   ! Warning
-     The functionality contained in this module exposes certain implementation
-     details of the language. As such, the API has no stability guarantees and
-     is subject to change as the Enso interpreter evolves.
-type Meta
-
-    ## UNSTABLE
-       ADVANCED
-
-       An Atom meta-representation.
-
-       Arguments:
-       - value: The value of the atom in the meta representation.
-    type Atom value
+   Arguments:
+   - value: The value of the atom in the meta representation.
+type Atom
+    Atom_Data value
```

Comment:
Is there a particular reason why we make every subtype of the `Meta` "class" a separate type?

It seems most intuitive to me that this should be an ADT: `Meta = Atom | Primitive | ...`.

I guess it is because different subtypes have different sets of functions available? I think we need to figure out a good guideline for such cases.

I don't think we should stall the integration with thinking how to resolve that - I do understand that you performed the transformations as to preserve the original semantics as closely as possible - so I think any changes will make sense to be done separately. Just noting this so we can discuss it for later.

Reply:
It seemed like these are useful as separate types, as we often know which will be the result.